### PR TITLE
Installed Operators UI updates

### PR DESCRIPTION
**Pull request by rebeccaalpert**
I updated the actions in kebab menu, removed filtering, and added a success icon to the "Copied" and "InstallSucceeded" status states.

Screen Shot 2019-06-05 at 4 36 39 PM

[I added a Jira issue](https://jira.coreos.com/browse/CONSOLE-1521) for adding "Edit Subscription" and "Uninstall Operator" to the kebab menu. We're going to have to do it as a follow-on since CSVs cannot currently link to Subscriptions. Per Alec, we can only link from Subscriptions to CSVs right now.

Fixes [CONSOLE-1462](https://jira.coreos.com/projects/CONSOLE/issues/CONSOLE-1462).
Fixes [OLM-903](https://jira.coreos.com/browse/OLM-903)

@openshift/team-ux-review, can you please review?

### CODE REVIEW

File: frontend/__tests__/components/operator-lifecycle-manager/clusterserviceversion.spec.tsx [hunk 1]
```diff
@@ -88,7 +88,7 @@ describe(ClusterServiceVersionRow.displayName, () => {
   it('renders column for app status', () => {
     const col = wrapper.find('.row').childAt(3);
 
-    expect(col.childAt(0).text()).toEqual(CSVConditionReason.CSVReasonInstallSuccessful);
+    expect(col.childAt(0).text()).toEqual(` ${CSVConditionReason.CSVReasonInstallSuccessful}`);
```
Comment (spadgett) [inline frontend/__tests__/components/operator-lifecycle-manager/clusterserviceversion.spec.tsx]: Does this work?

```suggestion
    expect(col.childAt(0).text()).toEqual(expect.stringContaining(CSVConditionReason.CSVReasonInstallSuccessful));
```

File: frontend/public/components/operator-lifecycle-manager/clusterserviceversion.tsx [hunk 2]
```diff
@@ -47,13 +47,14 @@ export const ClusterServiceVersionHeader: React.SFC = () => <ListHeader>
...
-const menuActions = [Kebab.factory.Edit, Kebab.factory.Delete];
+const menuActions = [Kebab.factory.Edit];
 
 export const ClusterServiceVersionRow = withFallback<ClusterServiceVersionRowProps>(({obj}) => {
   const route = `/k8s/ns/${obj.metadata.namespace}/${ClusterServiceVersionModel.plural}/${obj.metadata.name}`;
 
+  const statusString = _.get(obj, 'status.reason', ClusterServiceVersionPhase.CSVPhaseUnknown);
   const installStatus = obj.status && obj.status.phase !== ClusterServiceVersionPhase.CSVPhaseFailed
-    ? <span>{_.get(obj, 'status.reason', ClusterServiceVersionPhase.CSVPhaseUnknown)}</span>
+    ? <span>{(statusString === 'Copied' || statusString === 'InstallSucceeded') && <i aria-hidden="true" className="pficon pficon-ok" />} {statusString}</span>
```
Comment (spadgett) [inline frontend/public/components/operator-lifecycle-manager/clusterserviceversion.tsx]: This should use the `co-icon-and-text` markup. (Same for the existing icon below.)

File: frontend/public/components/operator-lifecycle-manager/clusterserviceversion.tsx [hunk 3]
```diff
@@ -102,14 +103,6 @@ export const ClusterServiceVersionsPage = connect(stateToProps)((props: ClusterS
     Installed Operators are represented by Cluster Service Versions within this namespace. For more information, see the <ExternalLink href="https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/architecture.md" text="Operator Lifecycle Manager documentation" />. Or create an Operator and Cluster Service Version using the <ExternalLink href="https://github.com/operator-framework/operator-sdk" text="Operator SDK" />.
   </p>;
 
-  const allFilterValues = [CSVConditionReason.CSVReasonInstallSuccessful, CSVConditionReason.CSVReasonCopied];
-  const rowFilters = [{
-    type: 'clusterserviceversion-status',
```
Comment (spadgett) [inline frontend/public/components/operator-lifecycle-manager/clusterserviceversion.tsx]: You need to remove the corresponding filter from list.tsx

File: frontend/__tests__/components/operator-lifecycle-manager/clusterserviceversion.spec.tsx (hunk 1 above)
Comment (rebeccaalpert) [reply frontend/__tests__/components/operator-lifecycle-manager/clusterserviceversion.spec.tsx]: I don't think we need to change the test here if I add the co-icon-and-text class.

File: frontend/public/components/factory/list.tsx [hunk 4]
```diff
@@ -111,12 +111,6 @@ const listFilters = {
     }
     return filters.selected.has(resource.kind);
   },
-  'clusterserviceversion-status': (filters, csv) => {
```
Comment (alecmerdler) [inline frontend/public/components/factory/list.tsx]: :+1:

File: frontend/public/components/operator-lifecycle-manager/clusterserviceversion.tsx [hunk 5]
```diff
@@ -47,14 +48,16 @@ export const ClusterServiceVersionHeader: React.SFC = () => <ListHeader>
   <ColHead className="col-lg-3 col-md-3 hidden-sm hidden-xs">Provided APIs</ColHead>
 </ListHeader>;
 
-const menuActions = [Kebab.factory.Edit, Kebab.factory.Delete];
+const menuActions = [Kebab.factory.Edit];
```
Comment (alecmerdler) [inline frontend/public/components/operator-lifecycle-manager/clusterserviceversion.tsx]: There's been a UX desire for the kebab options to include "full delete" (identical to [**Remove Subscription...** option from `Subscription` list view](https://github.com/openshift/console/blob/master/frontend/public/components/operator-lifecycle-manager/subscription.tsx#L36)). Can we include that in this PR?

Comment (rebeccaalpert) [reply frontend/public/components/operator-lifecycle-manager/clusterserviceversion.tsx]: Yeah, I wanted to add that, but it didn't look possible with what data we have right now.

Kebab factory uses this for deletion:
```
Delete: (kind, obj) => ({
    label: `Delete ${kind.label}`,
    callback: () => deleteModal({
      kind,
      resource: obj,
    }),
    accessReview: {
      group: kind.apiGroup,
      resource: kind.path,
      name: obj.metadata.name,
      namespace: obj.metadata.namespace,
      verb: 'delete',
    },
  }),
```

The deleteModal would require information like resource.metadata.name for the Subscription, which the CSV doesn't have access to. (Similar issue to the "Edit Subscription" action.)

Comment (rebeccaalpert) [reply frontend/public/components/operator-lifecycle-manager/clusterserviceversion.tsx]: Yeah, I tried this and it runs into the same issue as the "Edit Subscription" kebab option - we don't know the subscription's name.

Comment (spadgett) [reply frontend/public/components/operator-lifecycle-manager/clusterserviceversion.tsx]: If there's no simple way to add it now, we can make that a follow on.

Comment (spadgett) [reply frontend/public/components/operator-lifecycle-manager/clusterserviceversion.tsx]: Let's make sure there's an issue tracking it, though, so we don't forget.

Comment (rebeccaalpert) [reply frontend/public/components/operator-lifecycle-manager/clusterserviceversion.tsx]: It's already included in https://jira.coreos.com/browse/CONSOLE-1521.

**Pull request merged**